loader/loader.c

```diff
@@ -4918,6 +4947,58 @@ VkResult loader_create_instance_chain(const VkInstanceCreateInfo *pCreateInfo, c
         }
     }
 
+    // Make sure each layer requested by the application was actually loaded
+    for (uint32_t exp = 0; exp < inst->expanded_activated_layer_list.count; ++exp) {
+        struct loader_layer_properties *exp_layer_prop = &inst->expanded_activated_layer_list.list[exp];
+        bool found = false;
+        for (uint32_t act = 0; act < num_activated_layers; ++act) {
+            if (!strcmp(activated_layers[act].name, exp_layer_prop->info.layerName)) {
+                found = true;
+                break;
+            }
+        }
+        // If it wasn't found, we want to at least log an error.  However, if it was enabled by the application directly,
+        // we want to return a bad layer error.
+        if (!found) {
+            bool app_requested = false;
+            for (uint32_t act = 0; act < pCreateInfo->enabledLayerCount; ++act) {
+                if (!strcmp(pCreateInfo->ppEnabledLayerNames[act], exp_layer_prop->info.layerName)) {
+                    app_requested = true;
+                    break;
+                }
+            }
+            VkFlags log_flag = VULKAN_LOADER_LAYER_BIT;
+            char ending = '.';
+            if (app_requested) {
+                log_flag |= VULKAN_LOADER_ERROR_BIT;
+                ending = '!';
+            } else {
+                log_flag |= VULKAN_LOADER_INFO_BIT;
+            }
+            switch (exp_layer_prop->lib_status) {
+                case LOADER_LAYER_LIB_NOT_LOADED:
+                    loader_log(inst, log_flag, 0, "Requested layer %s was not loaded%c", exp_layer_prop->info.layerName, ending);
+                    break;
+                case LOADER_LAYER_LIB_ERROR_WRONG_BIT_TYPE: {
+                    loader_log(inst, log_flag, 0, "Requested layer %s was wrong bit-type%c", exp_layer_prop->info.layerName,
+                               ending);
+                    break;
+                }
+                case LOADER_LAYER_LIB_ERROR_FAILED_TO_LOAD:
+                    loader_log(inst, log_flag, 0, "Requested layer %s failed to load%c", exp_layer_prop->info.layerName, ending);
+                    break;
+                case LOADER_LAYER_LIB_SUCCESS_LOADED:
+                    // Needs to handle for compiler warnings but shouldn't get here.
+                    loader_log(inst, log_flag, 0, "Valid version of requested layer %s was loaded but also not found%c",
```

Comment:
Ah, I understand the intent of this message but it took me a while. Might be good to add more info about why this message is being reported. Should read kinda like an assertion statement.

```suggestion
                    // Shouldn't be able to reach this but if it is, best to report a debug 
                    loader_log(inst, log_flag, 0, "Shouldn't reach this. A valid version of requested layer %s was loaded but was not found in the list of activated layers%c",
```